Loadsys/Sniffs/NamingConventions/ValidPrivateProtectedFunctionNameSniff.php

```diff
@@ -98,7 +98,7 @@ protected function processTokenWithinScope(PHP_CodeSniffer_File $phpcsFile, $sta
 		if ($isPublic === true) {
 			if ($methodName[0] === '_') {
 				$error = 'Public method name "%s" must not be prefixed with underscore';
-				$phpcsFile->addError($error, $stackPtr, 'PublicWithUnderscore', $errorData);
+				$phpcsFile->addWarning($error, $stackPtr, 'PublicWithUnderscore', $errorData);
 				return;
```

Comment:
All PHP Codesniffer errors and warnings are on a numeric scale, varying only by the minimum "severity" value before a "failure" flips from a warning into an error. The docs explaining this are here: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#filtering-errors-and-warnings-based-on-severity

What that means is that if you lower the severity below the normal "error" threshold, it becomes a warning automatically. The `addError()` and `addWarning()` methods in the PHP code are just shortcuts to set the default severity values for errors and warnings respectively, like calling `Shell::err('msg')` instead of `Shell::out('<error>msg</error>')` in Cake.